libnymea-core/cloud/cloudtransport.cpp

```diff
@@ -0,0 +1,125 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ *                                                                         *
+ *  Copyright (C) 2018 Michael Zanetti <michael.zanetti@guh.io>            *
+ *  Copyright (C) 2018 Simon Stürz <simon.stuerz@guh.io>                   *
+ *                                                                         *
+ *  This file is part of nymea.                                            *
+ *                                                                         *
+ *  nymea is free software: you can redistribute it and/or modify          *
+ *  it under the terms of the GNU General Public License as published by   *
+ *  the Free Software Foundation, version 2 of the License.                *
+ *                                                                         *
+ *  nymea is distributed in the hope that it will be useful,               *
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of         *
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          *
+ *  GNU General Public License for more details.                           *
+ *                                                                         *
+ *  You should have received a copy of the GNU General Public License      *
+ *  along with nymea. If not, see <http://www.gnu.org/licenses/>.          *
+ *                                                                         *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#include "cloudtransport.h"
+#include "loggingcategories.h"
+
+#include "nymeacore.h"
+
+using namespace remoteproxyclient;
+
+namespace nymeaserver {
+
+CloudTransport::CloudTransport(const ServerConfiguration &config, QObject *parent):
+    TransportInterface(config, parent)
+{
+    m_proxyUrl = QUrl("wss://remoteproxy.nymea.io");
+}
+
+void CloudTransport::sendData(const QUuid &clientId, const QByteArray &data)
+{
+    qCDebug(dcCloudTraffic()) << "Sending data" << clientId << data;
+    foreach (const ConnectionContext &ctx, m_connections) {
+        if (ctx.clientId == clientId) {
+            ctx.proxyConnection->sendData(data);
+            return;
+        }
+    }
+    qCWarning(dcCloud()) << "Error sending data. No such clientId";
+}
+
+void CloudTransport::sendData(const QList<QUuid> &clientIds, const QByteArray &data)
+{
+    foreach (const QUuid &clientId, clientIds) {
+        sendData(clientId, data);
+    }
+}
+
+bool CloudTransport::startServer()
+{
+    qCDebug(dcCloud()) << "Started cloud transport";
```

Comment:
This is not implemented yet. Do we need it?

Reply:
Well, the TransportInterface requires this API, as the remote connection transport is in fact a client, not a server implementation, it doesn't need to do anything here. We could have an empty default implementation in NymeaTransportInterface but given that in virtually all other cases a transport interface needs to actually do something, I'd probably vote for having an empty implementation here instead of having the risk that a new transport implementation starts the server in the wrong place because the developer missed this method...

Really a matter of taste I guess...